utils/python/CIME/case_st_archive.py

```diff
@@ -249,8 +249,11 @@ def _archive_restarts(case, archive, archive_entry,
                 # need to do this before archiving restart files
                 histfiles_for_restart = get_histfiles_for_restarts(case, archive,
                                                                    archive_entry, restfile)
+
                 if datename_is_last and histfiles_for_restart:
-                    histfiles_savein_rundir = histfiles_for_restart
+                    for histfile in histfiles_for_restart:
```

Comment:
st_archiver was dropping files in multi-instance cases, this is a fix for that problem.